---
title: "Combining genetic data and DNA methylation in\n START project"
author: "Julia Romanowska"
date: "16-10-2020"
output:
  xaringan::moon_reader:
    css: xaringan-themer.css
    lib_dir: libs
    nature:
      highlightStyle: solarized-light
      highlightLines: true
      countIncrementalSlides: false
fig_width: 13
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, fig.width = 13, results = "hide",
                      message = FALSE)
options(htmltools.dir.version = FALSE)
library(xaringanthemer)
style_solarized_light()
xaringanExtra::use_clipboard()
xaringanExtra::use_fit_screen()
source("p.sim.R")
```

class: left, inverse, bottom

# OVERVIEW

## Ideas
## Examples
## Discussion

???


---

class: left, inverse, bottom

# IDEAS:


## meQTLs & PoO-effect
## X-chromosome analysis
## interaction effect: PoOxMe

---

class: left, inverse, bottom

## meQTLs

*involved: Julia, Miriam, H책kon, Anil, Rolv Terje*

---

## meQTLs - idea

* calculate meQTLs in parents only (cases and controls separately)
* check how the alleles are divided
* is there a difference between cases and controls?
* how about changes between parents and children? is there difference between case-families and control-families?
* calculate the same in children and check significance

---

**meQTL:** No

**ART-effect:** No
```{r}
p.sim(levs = rep("LLLL", 6))
```

---

**meQTL:** No

**ART-effect:** No
```{r}
p.sim(levs = rep("MMMM", 6))
```

---

**meQTL:** No

**ART-effect:** Yes
```{r}
p.sim(levs = c(rep("MMMM", 3), rep("HHHH", 3)))
```

---

**meQTL:** Yes

**ART-effect:** No
```{r}
p.sim(levs = c(rep("LHHH", 3), rep("LHHH", 3)))
```

---

**meQTL:** Yes

**ART-effect:** Yes (after fertilization?)
```{r}
p.sim(levs = c(rep("LHHH", 3), c("LHHH", "LHHH", "LLLL")))
```

---

**meQTL:** Yes

**ART-effect:** Yes

**POO ART-effect:** Yes (maternal line?)
```{r}
p.sim(levs = c(rep("LHHH", 3), c("LHHH", "LHHH", "LHLH")))
```

---

**meQTL:** Yes

**ART-effect:** Yes

**POO ART-effect:** Yes (paternal line?)
```{r}
p.sim(levs = c(rep("LHHH", 3), c("LHHH", "LHHH", "LLHH")))
```

---
## meQTLs - Discussion

* are meQTLs PoO-specific?    
*(if the ART procedures affect the methylation before fertilisation, then we should see the PoO-specific meQTLs)*
* what does it say that there is (no) difference between cases and controls?
* how likely it is that meQTLs are shared between child and parent?    
*(how do they work during the fertilisation/embryogenesis?)*

---

class: left, inverse, bottom

## X-chromosome analysis

*involved: Julia, H책kon N, H책kon G, Anil, Rolv Terje*

---

## X-chrom - idea

* X-EWAS    
*we need to find the right method*
* look specifically for imprinted regions?

---

class: left, inverse, bottom

## interaction effect: PoOxMe

*involved: Julia, Ellisif?, H책kon, Anil, Rolv Terje*

---

## PoOxMe

* use our new methodology!
* Haplin & HaplinMethyl
* how to treat the meQTLs found before?
