<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Combining genetic data and DNA methylation in START project</title>
    <meta charset="utf-8" />
    <meta name="author" content="Julia Romanowska" />
    <script src="libs/clipboard/clipboard.min.js"></script>
    <link href="libs/xaringanExtra-clipboard/xaringanExtra-clipboard.css" rel="stylesheet" />
    <script src="libs/xaringanExtra-clipboard/xaringanExtra-clipboard.js"></script>
    <script>window.xaringanExtraClipboard(null, {"button":"Copy Code","success":"Copied!","error":"Press Ctrl+C to Copy"})</script>
    <script src="libs/xaringanExtra_fit-screen/fit-screen.js"></script>
    <link rel="stylesheet" href="xaringan-themer.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Combining genetic data and DNA methylation in START project
### Julia Romanowska
### 16-10-2020

---




class: left, inverse, bottom

# OVERVIEW

## Ideas
## Examples
## Discussion

???


---

class: left, inverse, bottom

# IDEAS:


## meQTLs &amp; PoO-effect
## X-chromosome analysis
## interaction effect: PoOxMe

---

class: left, inverse, bottom

## meQTLs

*involved: Julia, Miriam, H책kon, Anil, Rolv Terje*

---

## meQTLs - idea

* calculate meQTLs in parents only (cases and controls separately)
* check how the alleles are divided
* is there a difference between cases and controls?
* how about changes between parents and children? is there difference between case-families and control-families?
* calculate the same in children and check significance

---

**meQTL:** No

**ART-effect:** No
![](gene-methyl_interact_JRomanowska_files/figure-html/unnamed-chunk-1-1.png)&lt;!-- --&gt;

---

**meQTL:** No

**ART-effect:** No
![](gene-methyl_interact_JRomanowska_files/figure-html/unnamed-chunk-2-1.png)&lt;!-- --&gt;

---

**meQTL:** No

**ART-effect:** Yes
![](gene-methyl_interact_JRomanowska_files/figure-html/unnamed-chunk-3-1.png)&lt;!-- --&gt;

---

**meQTL:** Yes

**ART-effect:** No
![](gene-methyl_interact_JRomanowska_files/figure-html/unnamed-chunk-4-1.png)&lt;!-- --&gt;

---

**meQTL:** Yes

**ART-effect:** Yes (after fertilization?)
![](gene-methyl_interact_JRomanowska_files/figure-html/unnamed-chunk-5-1.png)&lt;!-- --&gt;

---

**meQTL:** Yes

**ART-effect:** Yes

**POO ART-effect:** Yes (maternal line?)
![](gene-methyl_interact_JRomanowska_files/figure-html/unnamed-chunk-6-1.png)&lt;!-- --&gt;

---

**meQTL:** Yes

**ART-effect:** Yes

**POO ART-effect:** Yes (paternal line?)
![](gene-methyl_interact_JRomanowska_files/figure-html/unnamed-chunk-7-1.png)&lt;!-- --&gt;

---
## meQTLs - Discussion

* are meQTLs PoO-specific?    
*(if the ART procedures affect the methylation before fertilisation, then we should see the PoO-specific meQTLs)*
* what does it say that there is (no) difference between cases and controls?
* how likely it is that meQTLs are shared between child and parent?    
*(how do they work during the fertilisation/embryogenesis?)*

---

class: left, inverse, bottom

## X-chromosome analysis

*involved: Julia, H책kon N, H책kon G, Anil, Rolv Terje*

---

## X-chrom - idea

* X-EWAS    
*we need to find the right method*
* look specifically for imprinted regions?

---

class: left, inverse, bottom

## interaction effect: PoOxMe

*involved: Julia, Ellisif?, H책kon, Anil, Rolv Terje*

---

## PoOxMe

* use our new methodology!
* Haplin &amp; HaplinMethyl
* how to treat the meQTLs found before?
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "solarized-light",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
